### PR TITLE
Make kms-key-id optional for encrypted EBS volumes

### PR DESCRIPTION
**Pull request by superseb**
https://github.com/rancher/rancher/issues/9633

### CODE REVIEW

**Comment by superseb** [comment]
Needs e2e testing, will do that

File: package/ebs/rancher-ebs [hunk 1]
```diff
@@ -138,9 +138,14 @@ create() {
     local encrypted=${OPTS[encrypted]}
     local encrypted_option=""
     if [ "${encrypted}" == "true" ]; then
+      # This parameter is only required if you want to use a non-default CMK; if this parameter is not specified, the default CMK for EBS is used.
+      # The first time you create an encrypted volume in a region, a default CMK is created for you automatically.
+      # This key is used for Amazon EBS encryption unless you select a CMK that you created separately using AWS KMS.
       if [ -z "${OPTS[kmsKeyId]}" ]; then
-          print_error "kmsKeyId is required for an encrypted volume"
+          echo "No kmsKeyId specified, using default CMK"
```
Comment (StrongMonkey) [inline package/ebs/rancher-ebs]: You can't echo the message to the output, we use the output to unmarshall json result and return that to cattle. Remove the `echo` then everything should work fine.

File: package/ebs/rancher-ebs [hunk 2]
```diff
@@ -138,9 +138,14 @@ create() {
...
+          encrypted_option="--encrypted"
       else
+          echo "Using specified kmsKeyId"
```
Comment (StrongMonkey) [inline package/ebs/rancher-ebs]: same above

**Comment by superseb** [comment]
Test 1 without key:
```version: '2'
volumes:
  bar:
    driver_opts:
      ec2_az: eu-west-1c
      encrypted: 'true'
      size: '10'
    driver: rancher-ebs
services:
  foo:
    image: busybox
    stdin_open: true
    volumes:
    - bar:/var/lib/storage
```

Test 2 with key:

```version: '2'
volumes:
  bar:
    driver_opts:
      ec2_az: eu-west-1c
      encrypted: 'true'
      kmsKeyId: arn:aws:kms:etcetcetc
      size: '10'
    driver: rancher-ebs
services:
  foo:
    image: busybox
    stdin_open: true
    volumes:
    - bar:/var/lib/storage
```

**Comment by StrongMonkey** [comment]
LGTM

**Comment by cjellick** [comment]
LGTM

**Pull request merged**